README.md

```diff
@@ -282,6 +282,54 @@ $auth->verify($idToken, [
 [google-id-tokens]: https://developers.google.com/identity/sign-in/web/backend-auth
 [iap-id-tokens]: https://cloud.google.com/iap/docs/signed-headers-howto
 
+## Caching
+We currently have limited support for caching. Caching is enabled by passing a PSR-6 `CacheItemPoolInterface` instance to the constructor when instantiating the credentials.
+
+We offer some caching classes out of the box under the `Google\Auth\Cache` namespace.
+
+```php
+use Google\Auth\ApplicationDefaultCredentials;
+use Google\Auth\Cache\MemoryCacheItemPool;
+use GuzzleHttp\HandlerStack;
+
+// specify the path to your application credentials
+putenv('GOOGLE_APPLICATION_CREDENTIALS=/path/to/my/credentials.json');
+
+// Provide the ID token audience. This can be a Client ID associated with an IAP application
+//    $targetAudience = 'IAP_CLIENT_ID.apps.googleusercontent.com';
+$targetAudience = 'YOUR_ID_TOKEN_AUDIENCE';
+
+// Cache Instance
+$memoryCache = new MemoryCacheItemPool;
+
+// Create Middleware
+$middleware = ApplicationDefaultCredentials::getProxyIdTokenMiddleware($targetAudience, cache: $memoryCache);
```

Comment:
It's weird to highlight `getProxyIdTokenMiddleware`, as this is a relatively obscure method. I would recommend illustrating `ApplicationDefaultCredentials::getCredentials` or `getMiddleware`, as these are the most common method users will call.